### ISSUE TITLE
http https proxy + kubeconfig

### ISSUE THREAD

**Issue by dzirg44** [helm/helm#7921]
Output of `helm version`:
```
version.BuildInfo{Version:"v3.1.2", GitCommit:"d878d4d45863e42fd5cff6743294a11d28a9abce", GitTreeState:"clean", GoVersion:"go1.13.8"}
```
Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-26T06:16:15Z", GoVersion:"go1.14", Compiler:"gc", Platform:"darwin/amd64"}
```
kubectl works fine
```
HTTPS_PROXY=https://user:PASS@my.proxy.io:443 kubectl --kubeconfig=kubeconfig --insecure-skip-tls-verify
```
but helm doesn't work
i set up also env variables
```
HTTPS_PROXY=https://user:PASS@my.proxy.io:443
KUBECONFIG=kubectl
HTTP_PROXY=https://user:PASS@my.proxy.io:443
https_proxy=https://user:PASS@my.proxy.io:443
http_proxy=https://user:PASS@my.proxy.io:443
```
also i tried with
```
helm --kubeconfig kubeconfig list
```
but in every case i just received
`Error: Kubernetes cluster unreachable`

debug
```
I0414 20:11:32.700481    9794 round_trippers.go:423] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: helm/v0.0.0 (darwin/amd64) kubernetes/$Format" 'https://internalclustername:6443/version?timeout=32s'
I0414 20:11:32.812979    9794 round_trippers.go:443] GET https://internalclustename:6443/version?timeout=32s  in 112 milliseconds
I0414 20:11:32.813002    9794 round_trippers.go:449] Response Headers:
Error: Kubernetes cluster unreachable
```
is it possible to work through a proxy?

**Comment by technosophos** [comment]
I am having trouble reproducing. Helm simply uses the same HTTP client and env vars that `kubectl` does. And the HTTPS_PROXY code is built into the `net/http` library in Go, so even looking through the code I can't find how this could be happening in your case.

Perhaps somewhere the HTTP RoundTripper is being overridden. Or perhaps something changed in 1.18 that is impacting both `kubectl` and `helm`'s `master` branch (which is on 1.18).

**Comment by dzirg44** [comment]
I figured out what is going on. 
Previously with `helm2` we had
```proxyconnect tcp: x509: certificate signed by unknown authority```
but now we have just 
```
Kubernetes cluster unreachable
```
in my opinion it is not so obvious and I hope it will help somebody.

**Comment by technosophos** [comment]
Oh! That is very helpful. Thank you.

**Comment by technosophos** [comment]
It looks like @adamreese has already fixed this for Helm 3.2! I am marking this as closed.
